### DISCUSSION TITLE
How do I get my route tree based on file routing?

### DISCUSSION THREAD

**Discussion by Rocky268** [TanStack/router#2737] [General]
I configure my application based on file routing

I now have a scenario where I want to get all the routes to render the navigation menu on a specific page. How can I do this?

**Comment by schiller-manuel** [comment]
you can get access all routes a runtime using `router.flatRoutes`, `router.routesById` and `router.routesByPath`

**Comment by Atw-Lee** [comment]
I also need to show the menu，But how do set the name of the menu item。I don't want to write mappings。

**Reply by omridevk** [reply]
you can use the route context to add such information for example

**Reply by Atw-Lee** [reply]
thanks，used the staticData property
